docs: SIG Docs 2020 Annual Report (#5569)
kubernetes/community: sig-docs/annual-report-2021.md

Taylor Dolezal ([onlydole](https://github.com/onlydole)),
Tim Bannister ([sftim](https://github.com/sftim))

- [Operational](#operational)
- [Membership](#membership)
- [Current initiatives and project health](#current-initiatives-and-project-health)

## Operational

**How are you doing with operational tasks in [sig-governance.md](/committee-steering/governance/sig-governance.md)?**

  - **Is your README accurate? have a CONTRIBUTING.md file?**

    Yes. We also have `Contributing.md` file.
  - **All subprojects correctly mapped and listed in [sigs.yaml](/sig-list.md)?**

    Yes.
      
  - **What’s your meeting culture? Large/small, active/quiet, learnings? Meeting notes up to date?
    Are you keeping recordings up to date/trends in community members watching recordings?**
    
    Quite small (usually 5-10 people) and generally we try to make people feel comfortable to give their opinion.
    Meeting notes are up to date. We have recording of the meetings uploaded.

**How does the group get updates, reports, or feedback from subprojects? Are there any springing up or being retired?
Are OWNERS files up to date in these areas?**

> Yes it is.

Author: I was wondering if the localization one is actually a subproject @bradtopol I remember that this was founded last year

> Yes it is.

Reviewer: Subproject teams provide project updates at the SIG quarterly meetings or discuss changes at the weekly meetings.

> Yes it is.

Reviewer: It seems like the localization project is considered a subproject (maintains separate meetings like the Blog subproject).

> Yes it is.

Reviewer 2: I think it really depends on the long term goal. I know when the localization group was created, the intention was to unify all localization efforts and dissolve (making it a working group). However, I do see value in it continuing permanently as long as there's contributor support.

My "gut feeling" is that the localization group started as a WG and probably should be considered a subproject now. At the very least, it's worth calling out here.

> Yes it is.

Reviewer 3: > At the very least, it's worth calling out here.

+1

**How does the group get updates, reports, or feedback from Working Groups? Are there any springing up or being retired? 
Are OWNERS files up to date in these areas?**

Yes it is.

**When was your last monthly community-wide update? (provide link to deck and/or recording)**

<!-- TODO -->

## Membership

**Are all listed SIG leaders (chairs, tech leads, and subproject owners) active?**

Yes, they are.

**How do you measure membership? By mailing list members, OWNERs, or something else?**

We don't have an official way of measuring membership.

**How does the group measure reviewer and approver bandwidth? Do you need help in any area now? What are you doing about it?**

> PR are generally open for anyone to review. We have [PR wrangler rotation](https://github.com/kubernetes/website/wiki/PR-Wranglers).

Reviewer: nit: Any contributor can review a pull request. SIG Docs maintains a weekly [PR wrangler rotation](...) to help review, manage, and approve pull requests.

> In general each of the approver will taking the rotation. Under exceptional circumstances where there's lots of

Reviewer: nit: I think line 60 (and the included link) explains the PR rotation. I would rewrite line 61 as:

Each approver participates in the PR wrangler rotation. Approvers are supported during their rotation by other reviewers and approvers.

pressure for the wrangler, another approver usually will be willing to help.

> For technical reviews, other than SIG Docs Tech Leads, we also notify respective SIGs with necessary knowledge of the

Author: Yes, I need help on this area

> For technical reviews, other than SIG Docs Tech Leads, we also notify respective SIGs with necessary knowledge of the

Reviewer 4: Does SIG Docs need help with reviewer and approver bandwidth?

My gut feeling is that we don't have good enough measurements on that / aren't looking at the measurements enough to know. It sounds like we're missing a check, but if that's the accurate story then we should say so.

> For technical reviews, other than SIG Docs Tech Leads, we also notify respective SIGs with necessary knowledge of the

Reviewer: I don't think we have measured the types of contributions identified by SIG, per PR.

Could add something like:
SIG Docs receives contributions from community members representing different SIGs.
During the PR review process, SIG Docs reaches out to other SIGs when a pull request requires technical review.

technical aspect of the content.

**Is there a healthy onboarding and growth path for contributors in your SIG? 
What are some activities that the group does to encourage this? 
What programs are you participating in to grow contributors throughout the contributor ladder?**

There is no official onboarding process. However, the SIG is supportive of new contributors,
willing to walk through the contribution process and eager to see contributions completed.
Issues are created regularly and triaged. Issues are also created specifically for new 
contributors. There are issues labelled for the different contribution areas (web development, 
blog, first time contributors, task, concept, API, etc) which helps new contributors 

> identify how to get started.

Reviewer 4: From time to time we use the weekly meeting Zoom call to help new contributors get onboard.

**What programs do you participate in for new contributors?**

We have an in person session during the KubeCon.

**Does the group have contributors from multiple companies/affiliations? 
Can end users/companies contribute in some way that they currently are not?**

> <!-- TODO -->

Reviewer 4: Contributors from multiple companies: yes.

> Can end users/companies contribute in some way that they currently are not?

I suppose we're particular keen to hear about suggested improvements from people who are using Kubernetes in production and can see gaps in what's there.

## Current initiatives and project health

[ ] Please include links to KEPs and other supporting information that will be beneficial to multiple types of community members.

**What are initiatives that should be highlighted, lauded, shout out, that your group is proud of? 
Currently underway? What are some of the longer tail projects that your group is working on?**

> <!-- TODO -->

Reviewer 4: Longer term projects include
- co-ordination with WG naming for things like removing the word “slave” (and other problematic terms) from docs.
- publishing better information about releases https://github.com/kubernetes/website/issues/20293

**Year to date KEP work review: What’s now stable? Beta? Alpha? Road to alpha?**

We have ongoing KEP, but we don't have a deadline set on when it should be finished.

- [Doc policies for third party content](https://github.com/kubernetes/enhancements/tree/master/keps/sig-docs/1326-third-party-content-in-docs)

**What areas and/or subprojects does the group need the most help with?**

> <!-- TODO -->

Reviewer 4: My gut feel is that the blog project could use extra help.

Reviewer: Yes, could add the blog subproject.

Kaitlyn Barnard ([@kbarnard10](https://github.com/kbarnard10)),
Taylor Dolezal ([onlydole](https://github.com/onlydole)),
Tim Bannister ([sftim](https://github.com/sftim))

- [Operational](#operational)
- [Membership](#membership)
- [Current initiatives and project health](#current-initiatives-and-project-health)

## Operational

**How are you doing with operational tasks in [sig-governance.md](/committee-steering/governance/sig-governance.md)?**

  - **Is your README accurate? have a CONTRIBUTING.md file?**

    Yes. We also have `Contributing.md` file.
  - **All subprojects correctly mapped and listed in [sigs.yaml](/sig-list.md)?**

    Yes.
      
  - **What’s your meeting culture? Large/small, active/quiet, learnings? Meeting notes up to date?
    Are you keeping recordings up to date/trends in community members watching recordings?**
    
    Quite small (usually 5-10 people) and generally we try to make people feel comfortable to give their opinion.
    Meeting notes are up to date. We have recording of the meetings uploaded.
    
    We have a new sub-project, it's localization sub-project, where we conduct monthly meeting 

>     attended by all people from the localizatiin team. This sub-porjcet focusing on sharing any concerns,

Reviewer 2: ```suggestion
    attended by all people from the localization team. This sub-project focusing on sharing any concerns,
```

**Is there a healthy onboarding and growth path for contributors in your SIG? 
What are some activities that the group does to encourage this? 
What programs are you participating in to grow contributors throughout the contributor ladder?**

There is no official onboarding process. However, the SIG is supportive of new contributors,
willing to walk through the contribution process and eager to see contributions completed.
Issues are created regularly and triaged. Issues are also created specifically for new 
contributors. There are issues labelled for the different contribution areas (web development, 
blog, first time contributors, task, concept, API, etc) which helps new contributors 
identify how to get started.

From time to time we use the weekly meeting Zoom call to help new contributors get onboard.

**What programs do you participate in for new contributors?**

> We have an in person session during the KubeCon.

Reviewer 2: We used to host doc sprints, as things get "back to normal" it would be good to call out as an activity. Not sure if it needs to land in this report or not... The doc sprints acted as a way to onboard new contributors as well as tackle a lot of open issues.

> **Does the group have contributors from multiple companies/affiliations? 

Reviewer 2: ```suggestion
**Does the group have contributors from multiple companies/affiliations?**
```

Yes

> Can end users/companies contribute in some way that they currently are not?**

Reviewer 2: ```suggestion
**Can end users/companies contribute in some way that they currently are not?**
```

I suppose we're particular keen to hear about suggested improvements from people who are using Kubernetes in production and can see gaps in what's there.

## Current initiatives and project health

> [ ] Please include links to KEPs and other supporting information that will be beneficial to multiple types of community members.

Reviewer 2: ```suggestion
- [x] Please include links to KEPs and other supporting information that will be beneficial to multiple types of community members.
```

Longer term projects include

> - co-ordination with WG naming for things like removing the word “slave” (and other problematic terms) from docs.

Reviewer 2: ```suggestion
- coordination with WG naming for things like removing the word “slave” (and other problematic terms) from docs.
```

- publishing better information about releases kubernetes/website#20293

**What are initiatives that should be highlighted, lauded, shout out, that your group is proud of? 
Currently underway? What are some of the longer tail projects that your group is working on?**

> <!-- TODO -->

Reviewer 2: It would be good to call out the docsy theme work as well as the reference documentation generation work.

Currently underway? What are some of the longer tail projects that your group is working on?**

Shoutout to docsy theme work as well as the reference documentation generation :tada:.

**Year to date KEP work review: What’s now stable? Beta? Alpha? Road to alpha?**

We have ongoing KEP, but we don't have a deadline set on when it should be finished.

- [Doc policies for third party content](https://github.com/kubernetes/enhancements/tree/master/keps/sig-docs/1326-third-party-content-in-docs)
  There's also an umbrella issue written as well [kubernetes/website#20232](https://github.com/kubernetes/website/issues/20232).

**What areas and/or subprojects does the group need the most help with?**

- Blog subproject

**What's the average open days of a PR and Issue in your group? / what metrics does your group care about and/or measure?**

We have the information from [devstat](https://k8s.devstats.cncf.io/d/25/open-pr-age-by-repository-group?orgId=1&var-period=q&var-repogroup_name=SIG%20Docs&var-kind_name=All):

![Devstat 2020](devstats/devstats-quarter-2021.png)

For example, based on 10/01/2020 data, we have:

> Average number of opened PRs 1.23 K with the median opened PRs age 3 days 13 hours 52 minutes 10 seconds.

Reviewer 4: Maybe add that we informally review metrics quarterly. Also _ad hoc_ if anyone brings it up at a weekly meeting.

We aim to have < 100 open PRs for the English localization, and will take steps as needed if we see the figure climbing much above that.

Reviewer 3: that's a nice goal / ballpark benchmark to call out

- [Membership](#membership)
- [Current initiatives and project health](#current-initiatives-and-project-health)

## Operational

**How are you doing with operational tasks in [sig-governance.md](/committee-steering/governance/sig-governance.md)?**

  - **Is your README accurate? have a CONTRIBUTING.md file?**

    Yes. We also have `Contributing.md` file.
  - **All subprojects correctly mapped and listed in [sigs.yaml](/sig-list.md)?**

    Yes.
      
  - **What’s your meeting culture? Large/small, active/quiet, learnings? Meeting notes up to date?
    Are you keeping recordings up to date/trends in community members watching recordings?**
    
    Quite small (usually 5-10 people) and generally we try to make people feel comfortable to give their opinion.
    Meeting notes are up to date. We have recording of the meetings uploaded.

**How does the group get updates, reports, or feedback from subprojects? Are there any springing up or being retired?
Are OWNERS files up to date in these areas?**

<!-- TODO -->

**How does the group get updates, reports, or feedback from Working Groups? Are there any springing up or being retired? 
Are OWNERS files up to date in these areas?**

> <!-- TODO -->

Author: @sftim Once we have the thought of doing clean up of approver that's not active anymore on the shift, is this already done? Any specific procedure that we need to share?

Reviewer 4: I think we decided that OWNER files are up to date.

Author: Got it thanks 👍

> **When was your last monthly community-wide update? (provide link to deck and/or recording)**

Author: @jimangel Would you mind filling this part?

<!-- TODO -->

## Membership

**Are all listed SIG leaders (chairs, tech leads, and subproject owners) active?**

<!-- TODO -->

**How do you measure membership? By mailing list members, OWNERs, or something else?**

We don't have an official way of measuring membership.

**How does the group measure reviewer and approver bandwidth? Do you need help in any area now? What are you doing about it?**

PR are generally open for anyone to review. We have [PR wrangler rotation](https://github.com/kubernetes/website/wiki/PR-Wranglers).
In general each of the approver will taking the rotation. Under exceptional circumstances where there's lots of
pressure for the wrangler, another approver usually will be willing to help.

For technical reviews, other than SIG Docs Tech Leads, we also notify respective SIGs with necessary knowledge of the
technical aspect of the content.

**Is there a healthy onboarding and growth path for contributors in your SIG? 
What are some activities that the group does to encourage this? 
What programs are you participating in to grow contributors throughout the contributor ladder?**

> There is no official onboarding process.

Reviewer: Optional:
I would add that the SIG is supportive of new contributors, willing to walk through the contribution process and eager
to see contributions completed. Issues are created regularly and triaged. Issues are also created specifically for new contributors. There are issues labelled for the different contribution areas (web development, blog, first time contributors, task, concept, API, ...) which helps new contributors identify how to get started.

Author: Got it, I added it on the docs, thank you 👍

**What programs do you participate in for new contributors?**

> <!-- TODO -->

Author: This one usually will have a session on Kubecon, does this count?

Reviewer 4: We would do a docs session at Kubecon in person, yep.

Author: Updated, I put those in the docs

**Does the group have contributors from multiple companies/affiliations? 
Can end users/companies contribute in some way that they currently are not?**

<!-- TODO -->

## Current initiatives and project health

[ ] Please include links to KEPs and other supporting information that will be beneficial to multiple types of community members.

**What are initiatives that should be highlighted, lauded, shout out, that your group is proud of? 
Currently underway? What are some of the longer tail projects that your group is working on?**

<!-- TODO -->

**Year to date KEP work review: What’s now stable? Beta? Alpha? Road to alpha?**

> <!-- TODO -->

Author: For this one I don't think we have this, please correct me if I'm wrong

Reviewer 4: Cleanup for that is in progress, it's an ongoing process that doesn't target a specific release.

We could report on PRs linked to the umbrella issue if we want to?

Author: I think it may be beneficial to link to the umbrella issue then, do you have the link?

Reviewer 4: https://github.com/kubernetes/website/issues/20232

**What areas and/or subprojects does the group need the most help with?**

<!-- TODO -->

**What's the average open days of a PR and Issue in your group? / what metrics does your group care about and/or measure?**

> <!-- TODO -->

Author: Do we have something that can be used to track how many Issues and PR that available in general per day?

Reviewer 4: https://k8s.devstats.cncf.io/ is useful here
We usually look at total open PRs, total open issues (both for k/website specifically), and total open PRs labelled “language/en”.

Reviewer 4: Also see https://k8s.devstats.cncf.io/d/25/open-pr-age-by-repository-group?orgId=1&var-period=d7&var-repogroup_name=SIG%20Docs&var-kind_name=All

Author: I put the graph and the link to the graph, PTAL
